demos/_includes/DemoActionsUtil.php

```diff
@@ -131,7 +155,37 @@ public static function setupDemoActions(CountryLock $country): void
                 'args' => [
                     'age' => ['type' => 'integer', 'required' => true],
                     'city' => [],
-                    'gender' => ['type' => 'string', /* TODO 'values' => ['m' => 'Male', 'f' => 'Female'],*/ 'required' => true, 'default' => 'm'],
+                    'gender' => [
+                        'type' => 'string',
+                        'required' => true,
+                        'default' => 'm',
+                        'ui' => [
+                            'form' => [
+                                Dropdown::class, 'values' => ['m' => 'Male', 'f' => 'Female'],
+                            ],
+                        ],
+                    ],
+                ],
+                'fields' => [$country->fieldName()->iso3],
+                'callback' => function ($model, $age, $city, $gender) {
+                    $n = $gender === 'm' ? 'Mr.' : 'Mrs.';
+
+                    return 'Thank you ' . $n . ' at age ' . $age;
+                },
+                'preview' => function ($model, $age, $city, $gender) {
+                    return 'Gender = ' . $gender . ' / Age = ' . $age;
+                },
+            ]
+        );
+
+        $country->addUserAction(
+            'arg_using_model',
+            [
+                'caption' => 'Arg with Model',
+                'description' => 'Ask for Arguments set via a Data\Model. Allow usage of model validate() for your arguments',
+                'args' => [
+                    '__atk_model' => new ArgModel(new Array_([])),
+                    'extra' => ['type' => 'string'],
                 ],
                 'fields' => [$country->fieldName()->iso3],
                 'callback' => function ($model, $age, $city, $gender) {
```

Comment:
then define hintable phpdoc above `ArgModel` class and get the values here with magic props

with implicitly declared model (UA above, use ->get($country->fieldName()->age))

Reply:
Same here, $model is UserAction model not the argument model. Arguments are passed separately to the execute callback function.